lib/merit/rule.rb

```diff
@@ -7,19 +7,14 @@ class Rule
                   :multiple, :temporary, :score, :block, :category
 
     # Does this rule's condition block apply?
-    def applies?(target_obj = nil)
+    def applies?(target_obj = nil, current_user = nil)
       return true if block.nil? # no block given: always true
-
-      case block.arity
-      when 1 # Expects target object
-        if target_obj.present?
-          block.call(target_obj)
-        else
-          Rails.logger.warn '[merit] no target_obj found on Rule#applies?'
-          false
-        end
-      when 0
-        block.call
+      if target_obj.nil? && current_user.nil?
+        Rails.logger.warn '[merit] no target_obj
+                          or current_user found on Rule#applies?'
```

Comment:
Actually, these checks depend on the arity of the rule block. I think we'll need to rework a bit this class.